packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -0,0 +1,266 @@
+
+        function isSuperCall(x: Statement | typeof undefined): boolean {
+          return (
+            x !== undefined &&
+            x.type === "ExpressionStatement" &&
+            x.expression.type === "CallExpression" &&
+            x.expression.callee.type === "Super"
+          );
+        }
+
+        // Rest handled by Function visitor
+      },
+
+      TSParameterProperty(path) {
+        path.replaceWith(path.node.parameter);
+      },
+
+      ClassProperty(path) {
+        const { node } = path;
+        if (!node.value) {
+          path.remove();
+          return;
+        }
+
+        if (node.accessibility) node.accessibility = null;
+        if (node.abstract) node.abstract = null;
+        if (node.optional) node.optional = null;
+        if (node.typeAnnotation) node.typeAnnotation = null;
+      },
+
+      TSIndexSignature(path) {
+        path.remove();
+      },
+
+      ClassDeclaration(path) {
+        const { node } = path;
+        if (node.declare) {
+          path.remove();
+          return;
+        }
+        if (node.abstract) node.abstract = null;
+      },
+
+      Class({ node }) {
+        if (node.typeParameters) node.typeParameters = null;
+        if (node.superTypeParameters) node.superTypeParameters = null;
+        if (node.implements) node.implements = null;
+      },
+
+      Function({ node }) {
+        if (node.typeParameters) node.typeParameters = null;
+        if (node.returnType) node.returnType = null;
+
+        const p0 = node.params[0];
+        if (p0 && p0.type === "Identifier" && p0.name === "this") {
+          node.params = node.params.slice(1);
+        }
+      },
+
+      TSModuleDeclaration(path) {
+        if (!path.node.declare && path.node.id.type !== "StringLiteral") {
+          throw path.buildCodeFrameError("Namespaces are not supported.");
+        }
+        path.remove();
+      },
+
+      TSInterfaceDeclaration(path) {
+        path.remove();
+      },
+
+      TSTypeAliasDeclaration(path) {
+        path.remove();
+      },
+
+      TSEnumDeclaration(path) {
+        transpileEnum(path, t);
+      },
+
+      TSImportEqualsDeclaration(path) {
+        throw path.buildCodeFrameError("`import =` is not supported.");
+      },
+
+      TSTypeAssertion(path) {
+        path.replaceWith(path.node.expression);
+      },
+
+      TSAsExpression(path) {
+        path.replaceWith(path.node.expression);
+      },
+
+      TSNonNullExpression(path) {
+        path.replaceWith(path.node.expression);
+      },
+
+      CallExpression(path) {
+        path.node.typeParameters = null;
+      },
+
+      NewExpression(path) {
+        path.node.typeParameters = null;
+      },
+
+      TSExportAssignment(path) {
+        const moduleExports = t.memberExpression(
+          t.identifier("module"),
+          t.identifier("exports"),
+        );
+        path.replaceWith(
+          t.assignmentExpression("=", moduleExports, path.node.expression),
```

Comment:
Should we be generating an export statement instead, so they can transform into whatever their bundler expects?